clang/test/OpenMP/assumes_template_print.cpp

```diff
@@ -17,23 +17,23 @@ template <typename T>
 struct S {
   int a;
 // CHECK: template <typename T> struct S {
-// CHECK:     __attribute__((assume("ompx_global_assumption"))) void foo()     {
+// CHECK:     void foo() __attribute__((assume("ompx_global_assumption")))     {
```

Comment:
What causes these to swap sides?

Reply:
I think that attribute is added by `#pragma omp parallel` which appears after the definition of the function in terms of source locations. I thought we should probably filter out attributes that came from pragmas, however, it seems that we do not retain that information in the attribute and it says "gnu" rather than "pragma"...

Reply:
Is it perhaps set as "isImplicit"?  I wonder if we could filter out based on that?

Reply:
It is not implicit - we don't print them. I looked and could not find anything that could distinguish the ones that come from pragmas. I briefly looked to see if we can add this in but seems some more work to be able to ask `::isOMPAttr` in tablegen. Generally that's a good improvement but it has a different timeline to implement than that PR which is meant to be a hot fix...

Reply:
@alexey-bataev : Comments/concerns?